osversion/platform_compat_windows.go

```diff
@@ -0,0 +1,70 @@
+package osversion
+
+import (
+	"strconv"
+	"strings"
+)
+
+// List of stable ABI compliant
+// ltsc releases
+// Note: List must be sorted
+var compatLtscReleases = []uint{
+	V21H2Server,
+}
+
+// Checks for stable ABI compliant OS versions.
+// Every release after WS 2022 will support the previous
+// ltsc container image. Stable ABI is in preview mode
+// for windows 11 client.
+// Refer: https://learn.microsoft.com/en-us/virtualization/windowscontainers/deploy-containers/version-compatibility?tabs=windows-server-2022%2Cwindows-10#windows-server-host-os-compatibility
+func CanRun(hostOSVersion, ctrImgVersion string) bool {
```

Comment:
could we add some unit tests?  There is a lot going on here and be nice to have something to validate the logic

Reply:
Adding test cases needs to wait until containerd changes are made as the windows platform matcher still depends on exact version matching. Was planning to add tests in containerd and then create a separate PR in hcsshim to add test cases for the same

Reply:
Is there a reason for this to be in hcsshim vs containerd?  It feels strange to add functionality in one place and tests in another.  

since `func CanRun(hostOSVersion, ctrImgVersion string) bool` is taking strings and return bool, it is a testable function on its own outside of platform machting code (which would likely have its own tests?).

Reply:
Adding this functionality to hcsshim to make it easier/faster to maintain as we have more releases in the change and control the compatibility story on windows.
Also, hcsshim component is also going to have entire functionality tests in them but need to wait for containerd windows matcher function to actually call into CanRun() function for the tests to meaningfully test end to end scenarios.

I can add some tests to check CanRun()  for the time being with these changes.

Reply:
@jsturtevant @kevpar would you prefer to move this function out to containerd or keep it in hcsshim for the reasons mentioned above?

Reply:
> control the compatibility story on windows.

I think this is a valid reason for keeping it here.  It seems useful in other scenarios outside of containerd as well.

> I can add some tests to check CanRun() for the time being with these changes. 

I do think tests here is a good idea due to the number of conditions and branches in this function

Reply:
might change the name of `ctrImgVersion` as well to make it more universal

Reply:
Added tests to test the functionality of CanRun() function and also changed ctrImgVersion to guestOSVersion